build_runner/lib/src/generate/phase.dart

```diff
@@ -117,14 +117,14 @@ class InBuildPhase extends BuildPhase implements BuildAction {
     final settings = <String>[];
     if (isOptional) settings.add('optional');
     if (hideOutput) settings.add('hidden');
-    var result = '${builder.runtimeType} on $targetSources in $package';
+    var result = '$builderLabel on $targetSources in $package';
     if (settings.isNotEmpty) result += ' $settings';
     return result;
   }
 
   @override
   int get identity => _deepEquals.hash([
-        '${builder.runtimeType}',
+        builder.buildExtensions,
```

Comment:
We should probably also include the `builderLabel`?

Reply:
we may as well. Done